src/crosschainMap/crosschainMap.ts

```diff
@@ -47,6 +50,19 @@ export class CrosschainMap {
   public getTokenIDByAddress = ({ address, chainId }: TokenBaseParams) =>
     this.tokenMapByChain[chainId][address.toLowerCase()]?.id
 
+  public getCrosschainTokenBySymbol = ({
+    chainId,
+    symbol
+  }: {
+    chainId: SupportedChains
+    symbol: string
+  }): MappedToken | undefined => {
+    const tokenMap = this.tokenMapByChain[chainId]
+    if (!tokenMap) return
+    const tokens = Object.values(tokenMap)
+    return tokens.find((token) => token.symbol === symbol)
```

Comment:
you can create a Map of Maps then you don't need  to find again.

`new Map<ChainId, Map<string, TokenInfo>>()`

Reply:
I am not sure I follow, what you descript is the tokenMap object.
I don't follow what you mean. Please elaborate a bit or if you think this is both a bit troublesome to explain in writing and also important to fix, we can jump into a call.